sonar-xml-plugin/src/main/java/org/sonar/plugins/xml/highlighting/XMLHighlighting.java

```diff
@@ -0,0 +1,211 @@
+/*
+ * SonarQube XML Plugin
+ * Copyright (C) 2010 SonarSource
+ * sonarqube@googlegroups.com
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.sonar.plugins.xml.highlighting;
+
+import javax.xml.stream.Location;
+import javax.xml.stream.XMLInputFactory;
+import javax.xml.stream.XMLStreamConstants;
+import javax.xml.stream.XMLStreamException;
+import javax.xml.stream.XMLStreamReader;
+import java.io.ByteArrayInputStream;
+import java.io.File;
+import java.io.FileInputStream;
+import java.io.InputStream;
+import java.nio.charset.Charset;
+import java.nio.charset.StandardCharsets;
+import java.util.ArrayList;
+import java.util.List;
+
+public class XMLHighlighting {
+
+  private static final String XML_DECLARATION_TAG = "<?xml";
+
+  private static List<HighlightingData> highlighting;
+
+  private XMLHighlighting() {
+  }
+
+  public static List<HighlightingData> getHighlightingData(File file, Charset charset) throws Exception {
+    return getHighlightingData(new FileInputStream(file), new SourceFileOffsets(file, charset));
+  }
+
+  public static List<HighlightingData> getHighlightingData(String content) throws XMLStreamException {
+    InputStream inputStream = new ByteArrayInputStream(content.getBytes(StandardCharsets.UTF_8));
+    return getHighlightingData(inputStream, new SourceFileOffsets(content));
+  }
+
+  public static List<HighlightingData> getHighlightingData(InputStream inputStream, SourceFileOffsets sourceFileOffsets) throws XMLStreamException {
+    XMLStreamReader xmlReader = XMLInputFactory.newInstance().createXMLStreamReader(inputStream);
+    highlighting = new ArrayList<>();
+    highlightXmlDeclaration(sourceFileOffsets);
+
+    while (xmlReader.hasNext()) {
+      Location prevLocation = xmlReader.getLocation();
+      xmlReader.next();
+      int startOffset = sourceFileOffsets.startOffset(xmlReader.getLocation().getLineNumber(), xmlReader.getLocation().getColumnNumber());
```

Comment:
@pynicolas Indeed! Fixed in https://github.com/SonarCommunity/sonar-xml/commit/a81eb04415bc920476ec721a737ae8f097087448